lib/cli/index.js

```diff
@@ -2537,9 +2537,27 @@ export async function createNodejsBom(path, options) {
       // Parse package-lock.json if available
       const parsedList = await parsePkgLock(f, options);
       const dlist = parsedList.pkgList;
-      const tmpParentComponent = dlist.splice(0, 1)[0] || {};
-      tmpParentComponent.type = "application";
+      let tmpParentComponent = dlist.splice(0, 1)[0] || {};
       if (!Object.keys(parentComponent).length) {
+          const basePath = dirname(f);
+          const packageJsonF = join(basePath, "package.json");
+          if (existsSync(packageJsonF)) {
+              const pcs = await parsePkgJson(packageJsonF, true);
+              if (pcs.length) {
+                  tmpParentComponent = pcs[0];
+                  tmpParentComponent.type = "application";
+                  ppurl = new PackageURL(
+                      "npm",
+                      options.projectGroup || tmpParentComponent.group,
+                      "project-name" in options ? options.projectName : tmpParentComponent.name,
```

Comment:
This allows the use of the **--project-name** property

Reply:
Does ```options.projectName || tmpParentComponent.name``` work?

Reply:
No, your solution doesn’t work. options.projectName is always set to the folder name.
The only way I can check if --project-name was provided is by verifying whether project-name is contained in the options.

Reply:
Interesting observation! Will make a note of this. Thank you!